js/piwik.js

```diff
@@ -3735,6 +3738,10 @@ if (typeof window.Piwik !== 'object') {
                     return;
                 }
                 if (!configDoNotTrack && request) {
+                    if (configConsentRequired) { // send a consent=1 when explicit consent is given for the apache logs
```

Comment:
It already does that, ie, if `configHasConsent` is false, it never gets to this code. The case I was talking about was when configHasConsent = true because of opt-in, but configConsentRequired = false because the user doesn't care about GDPR.

Reply:
The opt-out feature I'm referring to btw is the new first party cookie one that uses the consent feature: https://github.com/matomo-org/matomo/blob/3.x-dev/js/piwik.js#L7333-L7346

Reply:
@diosmosis maybe you can add the check in the if statement (is not required but makes it easier to understand) and merge then?